sched/misc/assert.c

```diff
@@ -554,12 +554,22 @@ static void dump_deadlock(void)
 void _assert(FAR const char *filename, int linenum,
              FAR const char *msg, FAR void *regs)
 {
-  FAR struct tcb_s *rtcb = running_task();
+  FAR struct tcb_s *rtcb = current_task();
```

Comment:
This change I'm not quite sure about. running_task() seems to give the wrong tcb for me, but fetching the ready-to-run head (via current_task()) always results in the correct tcb.

Reply:
the panic may happen in the interrupt handler, in this case current_task point to the new high priority task which may different from the interrupted task(running_task). running_task is always same as current_task in the thread context, but running_task point to the interrupted task during interrupt handling, utils the interrupt process finish. At that point, running_task will sync with current_task again.
So, running_task always point to the running task, but ccurrent_task may point to the task which will run after finishing the inerrupt process.

Reply:
Ok thanks for the explanation. There is some other bug then, because when e.g. a page fault from user space occurs, running_task() never points to the user space thread that faulted but rather to a high priority kernel task (maybe a context switch is happening too). This info to me is useless, which is why I changed to "current_task" which gives the correct user space thread.

Need to find what the issue is, reverting this change.